Matveibos/hometask-4/my_array_rotate.rb

```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+require 'benchmark'
+
+# the module extends the class Array
+module A
+  def lab_rotate(num = -1)
+    arr = clone
+    separator = -num % arr.size
+    if separator.zero?
+      self
+    else
+      section1 = arr[0..separator - 1]
+      section2 = arr[separator..arr.size]
+      section2 + section1
+    end
```

Comment:
I see a lot of similarities between `lab_rotate` and `lab_rotate!` methods. Can we move similar parts of code into a separate method?